es-app/src/guis/GuiSlideshowScreensaverOptions.cpp

```diff
@@ -0,0 +1,122 @@
+#include "guis/GuiSlideshowScreensaverOptions.h"
+#include "Window.h"
+#include "Settings.h"
+#include "views/ViewController.h"
+
+#include "components/SwitchComponent.h"
+#include "components/SliderComponent.h"
+#include "components/TextComponent.h"
+#include "components/OptionListComponent.h"
+#include "components/MenuComponent.h"
+#include "guis/GuiMsgBox.h"
+#include "guis/GuiTextEditPopup.h"
+#include "PowerSaver.h"
+
+GuiSlideshowScreensaverOptions::GuiSlideshowScreensaverOptions(Window* window, const char* title) : GuiScreensaverOptions(window, title)
+{
+	ComponentListRow row;
+
+	// image duration (seconds)
+	auto sss_image_sec = std::make_shared<SliderComponent>(mWindow, 3.f, 45.f, 1.f, "s");
+	sss_image_sec->setValue((float)(Settings::getInstance()->getInt("ScreenSaverSwapImageTimeout") / (1000)));
+	addWithLabel(row, "SWAP IMAGE AFTER (SECS)", sss_image_sec);
+	addSaveFunc([sss_image_sec] {
+		int playNextTimeout = (int)round(sss_image_sec->getValue()) * (1000);
+		Settings::getInstance()->setInt("ScreenSaverSwapImageTimeout", playNextTimeout);
+		PowerSaver::updateTimeouts();
+	});
+
+	// stretch
+	auto sss_stretch = std::make_shared<SwitchComponent>(mWindow);
+	sss_stretch->setState(Settings::getInstance()->getBool("SlideshowScreenSaverStretch"));
+	addWithLabel(row, "STRETCH IMAGES", sss_stretch);
+	addSaveFunc([sss_stretch] {
+		Settings::getInstance()->setBool("SlideshowScreenSaverStretch", sss_stretch->getState());
+	});
+
+	// background audio file
+	auto sss_bg_audio_file = std::make_shared<TextComponent>(mWindow);
```

Comment:
You forgot to specify the font colour in parameters. This causes the location to be black instead of the standard 777. Use it like this:
`auto sss_bg_audio_file = std::make_shared<TextComponent>(mWindow, "", Font::get(FONT_SIZE_SMALL), 0x777777FF);`

Font can be small, its fine as location has a full field.